table.cc

```diff
@@ -2344,3 +2346,39 @@ table::as_mutation_source_excluding(std::vector<sstables::shared_sstable>& ssts)
         return this->make_reader_excluding_sstables(std::move(s), std::move(permit), ssts, range, slice, pc, std::move(trace_state), fwd, fwd_mr);
     });
 }
+
+future<> table::update_repair_time(dht::token_range range, gc_clock::time_point repair_time) {
+    _repair_time_map += std::make_pair(locator::token_metadata::range_to_interval(range), repair_time);
+    return make_ready_future<>();
+}
+
+gc_clock::time_point
+table::get_repair_based_tombstone_gc_before(const dht::decorated_key& dk) const {
+    // FIXME: Make propagation_delay tunable
+    constexpr auto propagation_delay = std::chrono::seconds(3600);
```

Comment:
Ah, the slack is already there.

Reply:
Yes. The following patch has made propagation_delay configurable.